go/common/headers.go

```diff
@@ -47,6 +48,42 @@ type BatchHeader struct {
 	CrossChainTree                SerializedCrossChainTree              `json:"crossChainTree"`          // Those are the leafs of the merkle tree hashed for privacy. Necessary for clients to be able to build proofs as they have no access to all transactions in a batch or their receipts.
 }
 
+func ConvertBatchHeaderToHeader(batchHeader *BatchHeader) *types.Header {
```

Comment:
we already have this in: `gethEncodingServiceImpl.CreateEthHeaderForBatch` where it's also cached

Reply:
This needs to be used by the authenticated client which has no access to those services. I'm not sure why the service is required for this conversion tho? Caching?